app/models/plugins/thermometer.rb

```diff
@@ -28,4 +29,57 @@ def liquid_data
   def name
     self.class.name.demodulize
   end
+
+  def update_goal
+    self.goal = self.determine_next_goal
+    self.save
```

Comment:
Forget my last comment. You're incrementing a number value, so use increment:

`increment!(:goal, determine_next_goal)`

It'll skip the setter (goes straight to attribute), and any validation checks on save.